src/expression.c

```diff
@@ -6961,6 +6961,25 @@ Expression *FileExp::semantic(Scope *sc)
     se = se->toUTF8(sc);
     name = (char *)se->string;
 
+    if (name[0] == ':')
```

Comment:
magicport/ddmd is a direct translation of the C++ code, so it uses C strings and `const(char)*`.